src/widget/widget.js

```diff
@@ -128,6 +130,51 @@ export default class Widget extends Plugin {
 
 				this._setSelectionOverElement( objectElement );
 			} );
+
+			return true;
+		}
+	}
+
+	/**
+	 * Handles arrow keys.
+	 *
+	 * @param {Boolean} isForward Set to true if arrow key should be handled in forward direction.
+	 * @returns {Boolean|undefined} Returns `true` if keys were handled correctly.
+	 */
+	_handleArrowKeys( isForward ) {
+		const modelDocument = this.editor.document;
+		const schema = modelDocument.schema;
+		const modelSelection = modelDocument.selection;
+		const objectElement = getSelectedElement( modelSelection );
+
+		// if object element is selected.
+		if ( objectElement && schema.objects.has( objectElement.name ) ) {
```

Comment:
That's a valid point. We can make a discussion about it when we will extracting widgets functionalities from image plugin.